configure-ironic.sh

```diff
@@ -2,6 +2,10 @@
 
 . /bin/ironic-common.sh
 
+USE_HTTP_BASIC=${USE_HTTP_BASIC:-false}
```

Comment:
I remember discussing it, but I don't remember the outcome: I'd prefer we remove this flag and use the non-empty value of IRONIC_HTTP_BASIC_USERNAME instead.